File: CodeSamples/php/SwaggerClient-php/lib/Model/CreditCardTransactionSettings.php
<?php

namespace Swagger\Client\Model;

use \ArrayAccess;
use \Swagger\Client\ObjectSerializer;

class CreditCardTransactionSettings implements ModelInterface, ArrayAccess
{
    const DISCRIMINATOR = null;

    protected static $swaggerModelName = 'CreditCardTransactionSettings';

    protected static $swaggerTypes = [
        'cvv' => '\Swagger\Client\Model\FieldInfo',
        'contact_name' => '\Swagger\Client\Model\FieldInfo',
        'contact_title' => '\Swagger\Client\Model\FieldInfo',
        'email_address' => '\Swagger\Client\Model\FieldInfo',
        'phone' => '\Swagger\Client\Model\FieldInfo',
        'fax' => '\Swagger\Client\Model\FieldInfo',
        'billing_address' => '\Swagger\Client\Model\FieldInfo',
        'billing_address2' => '\Swagger\Client\Model\FieldInfo',
        'billing_city' => '\Swagger\Client\Model\FieldInfo',
        'billing_state' => '\Swagger\Client\Model\FieldInfo',
        'billing_zip' => '\Swagger\Client\Model\FieldInfo',
        'po_number' => '\Swagger\Client\Model\FieldInfo',
        'invoice_number' => '\Swagger\Client\Model\FieldInfo',
        'business_name' => '\Swagger\Client\Model\FieldInfo',
        'customer_number' => '\Swagger\Client\Model\FieldInfo',
        'reference_note' => '\Swagger\Client\Model\FieldInfo',
        'sales_tax' => '\Swagger\Client\Model\FieldInfo',
        'tax_exempt' => '\Swagger\Client\Model\FieldInfo',
        'card_tracks' => '\Swagger\Client\Model\FieldInfo',
        'card_zip' => '\Swagger\Client\Model\FieldInfo',
        'card_address' => '\Swagger\Client\Model\FieldInfo',
        'item_description' => 'string',
        'default_transaction_type' => '\Swagger\Client\Model\TransactionType'
    ];

    protected static $swaggerFormats = [
        'cvv' => null,
        'contact_name' => null,
        'contact_title' => null,
        'email_address' => null,
        'phone' => null,
        'fax' => null,
        'billing_address' => null,
        'billing_address2' => null,
        'billing_city' => null,
        'billing_state' => null,
        'billing_zip' => null,
        'po_number' => null,
        'invoice_number' => null,
        'business_name' => null,
        'customer_number' => null,
        'reference_note' => null,
        'sales_tax' => null,
        'tax_exempt' => null,
        'card_tracks' => null,
        'card_zip' => null,
        'card_address' => null,
        'item_description' => null,
        'default_transaction_type' => null
    ];

    public static function swaggerTypes()
    {
        return self::$swaggerTypes;
    }

    public static function swaggerFormats()
    {
        return self::$swaggerFormats;
    }

    protected static $attributeMap = [
        'cvv' => 'CVV',
        'contact_name' => 'ContactName',
        'contact_title' => 'ContactTitle',
        'email_address' => 'EmailAddress',
        'phone' => 'Phone',
        'fax' => 'Fax',
        'billing_address' => 'BillingAddress',
        'billing_address2' => 'BillingAddress2',
        'billing_city' => 'BillingCity',
        'billing_state' => 'BillingState',
        'billing_zip' => 'BillingZip',
        'po_number' => 'PONumber',
        'invoice_number' => 'InvoiceNumber',
        'business_name' => 'BusinessName',
        'customer_number' => 'CustomerNumber',
        'reference_note' => 'ReferenceNote',
        'sales_tax' => 'SalesTax',
        'tax_exempt' => 'TaxExempt',
        'card_tracks' => 'CardTracks',
        'card_zip' => 'CardZip',
        'card_address' => 'CardAddress',
        'item_description' => 'ItemDescription',
        'default_transaction_type' => 'DefaultTransactionType'
    ];

    protected static $setters = [
        'cvv' => 'setCvv',
        'contact_name' => 'setContactName',
        'contact_title' => 'setContactTitle',
        'email_address' => 'setEmailAddress',
        'phone' => 'setPhone',
        'fax' => 'setFax',
        'billing_address' => 'setBillingAddress',
        'billing_address2' => 'setBillingAddress2',
        'billing_city' => 'setBillingCity',
        'billing_state' => 'setBillingState',
        'billing_zip' => 'setBillingZip',
        'po_number' => 'setPoNumber',
        'invoice_number' => 'setInvoiceNumber',
        'business_name' => 'setBusinessName',
        'customer_number' => 'setCustomerNumber',
        'reference_note' => 'setReferenceNote',
        'sales_tax' => 'setSalesTax',
        'tax_exempt' => 'setTaxExempt',
        'card_tracks' => 'setCardTracks',
        'card_zip' => 'setCardZip',
        'card_address' => 'setCardAddress',
        'item_description' => 'setItemDescription',
        'default_transaction_type' => 'setDefaultTransactionType'
    ];

    protected static $getters = [
        'cvv' => 'getCvv',
        'contact_name' => 'getContactName',
        'contact_title' => 'getContactTitle',
        'email_address' => 'getEmailAddress',
        'phone' => 'getPhone',
        'fax' => 'getFax',
        'billing_address' => 'getBillingAddress',
        'billing_address2' => 'getBillingAddress2',
        'billing_city' => 'getBillingCity',
        'billing_state' => 'getBillingState',
        'billing_zip' => 'getBillingZip',
        'po_number' => 'getPoNumber',
        'invoice_number' => 'getInvoiceNumber',
        'business_name' => 'getBusinessName',
        'customer_number' => 'getCustomerNumber',
        'reference_note' => 'getReferenceNote',
        'sales_tax' => 'getSalesTax',
        'tax_exempt' => 'getTaxExempt',
        'card_tracks' => 'getCardTracks',
        'card_zip' => 'getCardZip',
        'card_address' => 'getCardAddress',
        'item_description' => 'getItemDescription',
        'default_transaction_type' => 'getDefaultTransactionType'
    ];

    public static function attributeMap()
    {
        return self::$attributeMap;
    }

    public static function setters()
    {
        return self::$setters;
    }

    public static function getters()
    {
        return self::$getters;
    }

    public function getModelName()
    {
        return self::$swaggerModelName;
    }

    protected $container = [];

    public function __construct(array $data = null)
    {
        $this->container['cvv'] = isset($data['cvv']) ? $data['cvv'] : null;
        $this->container['contact_name'] = isset($data['contact_name']) ? $data['contact_name'] : null;
        $this->container['contact_title'] = isset($data['contact_title']) ? $data['contact_title'] : null;
        $this->container['email_address'] = isset($data['email_address']) ? $data['email_address'] : null;
        $this->container['phone'] = isset($data['phone']) ? $data['phone'] : null;
        $this->container['fax'] = isset($data['fax']) ? $data['fax'] : null;
        $this->container['billing_address'] = isset($data['billing_address']) ? $data['billing_address'] : null;
        $this->container['billing_address2'] = isset($data['billing_address2']) ? $data['billing_address2'] : null;
        $this->container['billing_city'] = isset($data['billing_city']) ? $data['billing_city'] : null;
        $this->container['billing_state'] = isset($data['billing_state']) ? $data['billing_state'] : null;
        $this->container['billing_zip'] = isset($data['billing_zip']) ? $data['billing_zip'] : null;
        $this->container['po_number'] = isset($data['po_number']) ? $data['po_number'] : null;
        $this->container['invoice_number'] = isset($data['invoice_number']) ? $data['invoice_number'] : null;
        $this->container['business_name'] = isset($data['business_name']) ? $data['business_name'] : null;
        $this->container['customer_number'] = isset($data['customer_number']) ? $data['customer_number'] : null;
        $this->container['reference_note'] = isset($data['reference_note']) ? $data['reference_note'] : null;
        $this->container['sales_tax'] = isset($data['sales_tax']) ? $data['sales_tax'] : null;
        $this->container['tax_exempt'] = isset($data['tax_exempt']) ? $data['tax_exempt'] : null;
        $this->container['card_tracks'] = isset($data['card_tracks']) ? $data['card_tracks'] : null;
        $this->container['card_zip'] = isset($data['card_zip']) ? $data['card_zip'] : null;
        $this->container['card_address'] = isset($data['card_address']) ? $data['card_address'] : null;
        $this->container['item_description'] = isset($data['item_description']) ? $data['item_description'] : null;
        $this->container['default_transaction_type'] = isset($data['default_transaction_type']) ? $data['default_transaction_type'] : null;
    }

    public function listInvalidProperties()
    {
        $invalidProperties = [];

        return $invalidProperties;
    }

    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }


    public function getCvv()
    {
        return $this->container['cvv'];
    }

    public function setCvv($cvv)
    {
        $this->container['cvv'] = $cvv;

        return $this;
    }

    public function getContactName()
    {
        return $this->container['contact_name'];
    }

    public function setContactName($contact_name)
    {
        $this->container['contact_name'] = $contact_name;

        return $this;
    }

    public function getContactTitle()
    {
        return $this->container['contact_title'];
    }

    public function setContactTitle($contact_title)
    {
        $this->container['contact_title'] = $contact_title;

        return $this;
    }

    public function getEmailAddress()
    {
        return $this->container['email_address'];
    }

    public function setEmailAddress($email_address)
    {
        $this->container['email_address'] = $email_address;

        return $this;
    }

    public function getPhone()
    {
        return $this->container['phone'];
    }

    public function setPhone($phone)
    {
        $this->container['phone'] = $phone;

        return $this;
    }

    public function getFax()
    {
        return $this->container['fax'];
    }

    public function setFax($fax)
    {
        $this->container['fax'] = $fax;

        return $this;
    }

    public function getBillingAddress()
    {
        return $this->container['billing_address'];
    }

    public function setBillingAddress($billing_address)
    {
        $this->container['billing_address'] = $billing_address;

        return $this;
    }

    public function getBillingAddress2()
    {
        return $this->container['billing_address2'];
    }

    public function setBillingAddress2($billing_address2)
    {
        $this->container['billing_address2'] = $billing_address2;

        return $this;
    }

    public function getBillingCity()
    {
        return $this->container['billing_city'];
    }

    public function setBillingCity($billing_city)
    {
        $this->container['billing_city'] = $billing_city;

        return $this;
    }

    public function getBillingState()
    {
        return $this->container['billing_state'];
    }

    public function setBillingState($billing_state)
    {
        $this->container['billing_state'] = $billing_state;

        return $this;
    }

    public function getBillingZip()
    {
        return $this->container['billing_zip'];
    }

    public function setBillingZip($billing_zip)
    {
        $this->container['billing_zip'] = $billing_zip;

        return $this;
    }

    public function getPoNumber()
    {
        return $this->container['po_number'];
    }

    public function setPoNumber($po_number)
    {
        $this->container['po_number'] = $po_number;

        return $this;
    }

    public function getInvoiceNumber()
    {
        return $this->container['invoice_number'];
    }

    public function setInvoiceNumber($invoice_number)
    {
        $this->container['invoice_number'] = $invoice_number;

        return $this;
    }

    public function getBusinessName()
    {
        return $this->container['business_name'];
    }

    public function setBusinessName($business_name)
    {
        $this->container['business_name'] = $business_name;

        return $this;
    }

    public function getCustomerNumber()
    {
        return $this->container['customer_number'];
    }

    public function setCustomerNumber($customer_number)
    {
        $this->container['customer_number'] = $customer_number;

        return $this;
    }

    public function getReferenceNote()
    {
        return $this->container['reference_note'];
    }

    public function setReferenceNote($reference_note)
    {
        $this->container['reference_note'] = $reference_note;

        return $this;
    }

    public function getSalesTax()
    {
        return $this->container['sales_tax'];
    }

    public function setSalesTax($sales_tax)
    {
        $this->container['sales_tax'] = $sales_tax;

        return $this;
    }

    public function getTaxExempt()
    {
        return $this->container['tax_exempt'];
    }

    public function setTaxExempt($tax_exempt)
    {
        $this->container['tax_exempt'] = $tax_exempt;

        return $this;
    }

    public function getCardTracks()
    {
        return $this->container['card_tracks'];
    }

    public function setCardTracks($card_tracks)
    {
        $this->container['card_tracks'] = $card_tracks;

        return $this;
    }

    public function getCardZip()
    {
        return $this->container['card_zip'];
    }

    public function setCardZip($card_zip)
    {
        $this->container['card_zip'] = $card_zip;

        return $this;
    }

    public function getCardAddress()
    {
        return $this->container['card_address'];
    }

    public function setCardAddress($card_address)
    {
        $this->container['card_address'] = $card_address;

        return $this;
    }

    public function getItemDescription()
    {
        return $this->container['item_description'];
    }

    public function setItemDescription($item_description)
    {
        $this->container['item_description'] = $item_description;

        return $this;
    }

    public function getDefaultTransactionType()
    {
        return $this->container['default_transaction_type'];
    }

    public function setDefaultTransactionType($default_transaction_type)
    {
        $this->container['default_transaction_type'] = $default_transaction_type;

        return $this;
    }
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    public function offsetGet($offset)
    {
        return isset($this->container[$offset]) ? $this->container[$offset] : null;
    }

    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }

    public function __toString()
    {
        if (defined('JSON_PRETTY_PRINT')) { // use JSON pretty print
            return json_encode(
                ObjectSerializer::sanitizeForSerialization($this),
                JSON_PRETTY_PRINT
            );
        }

        return json_encode(ObjectSerializer::sanitizeForSerialization($this));
    }
}

File: CodeSamples/python/swagger_client/models/credit_card_transaction_settings.py
# coding: utf-8

class CreditCardTransactionSettings(object):
    swagger_types = {
        'cvv': 'FieldInfo',
        'contact_name': 'FieldInfo',
        'contact_title': 'FieldInfo',
        'email_address': 'FieldInfo',
        'phone': 'FieldInfo',
        'fax': 'FieldInfo',
        'billing_address': 'FieldInfo',
        'billing_address2': 'FieldInfo',
        'billing_city': 'FieldInfo',
        'billing_state': 'FieldInfo',
        'billing_zip': 'FieldInfo',
        'po_number': 'FieldInfo',
        'invoice_number': 'FieldInfo',
        'business_name': 'FieldInfo',
        'customer_number': 'FieldInfo',
        'reference_note': 'FieldInfo',
        'sales_tax': 'FieldInfo',
        'tax_exempt': 'FieldInfo',
        'card_tracks': 'FieldInfo',
        'card_zip': 'FieldInfo',
        'card_address': 'FieldInfo',
        'item_description': 'str',
        'default_transaction_type': 'TransactionType'
    }

    attribute_map = {
        'cvv': 'CVV',
        'contact_name': 'ContactName',
        'contact_title': 'ContactTitle',
        'email_address': 'EmailAddress',
        'phone': 'Phone',
        'fax': 'Fax',
        'billing_address': 'BillingAddress',
        'billing_address2': 'BillingAddress2',
        'billing_city': 'BillingCity',
        'billing_state': 'BillingState',
        'billing_zip': 'BillingZip',
        'po_number': 'PONumber',
        'invoice_number': 'InvoiceNumber',
        'business_name': 'BusinessName',
        'customer_number': 'CustomerNumber',
        'reference_note': 'ReferenceNote',
        'sales_tax': 'SalesTax',
        'tax_exempt': 'TaxExempt',
        'card_tracks': 'CardTracks',
        'card_zip': 'CardZip',
        'card_address': 'CardAddress',
        'item_description': 'ItemDescription',
        'default_transaction_type': 'DefaultTransactionType'
    }

    def __init__(self, cvv=None, contact_name=None, contact_title=None, email_address=None, phone=None, fax=None, billing_address=None, billing_address2=None, billing_city=None, billing_state=None, billing_zip=None, po_number=None, invoice_number=None, business_name=None, customer_number=None, reference_note=None, sales_tax=None, tax_exempt=None, card_tracks=None, card_zip=None, card_address=None, item_description=None, default_transaction_type=None):  # noqa: E501
        """CreditCardTransactionSettings - a model defined in Swagger"""  # noqa: E501
        self._cvv = None
        self._contact_name = None
        self._contact_title = None
        self._email_address = None
        self._phone = None
        self._fax = None
        self._billing_address = None
        self._billing_address2 = None
        self._billing_city = None
        self._billing_state = None
        self._billing_zip = None
        self._po_number = None
        self._invoice_number = None
        self._business_name = None
        self._customer_number = None
        self._reference_note = None
        self._sales_tax = None
        self._tax_exempt = None
        self._card_tracks = None
        self._card_zip = None
        self._card_address = None
        self._item_description = None
        self._default_transaction_type = None
        self.discriminator = None
        if cvv is not None:
            self.cvv = cvv
        if contact_name is not None:
            self.contact_name = contact_name
        if contact_title is not None:
            self.contact_title = contact_title
        if email_address is not None:
            self.email_address = email_address
        if phone is not None:
            self.phone = phone
        if fax is not None:
            self.fax = fax
        if billing_address is not None:
            self.billing_address = billing_address
        if billing_address2 is not None:
            self.billing_address2 = billing_address2
        if billing_city is not None:
            self.billing_city = billing_city
        if billing_state is not None:
            self.billing_state = billing_state
        if billing_zip is not None:
            self.billing_zip = billing_zip
        if po_number is not None:
            self.po_number = po_number
        if invoice_number is not None:
            self.invoice_number = invoice_number
        if business_name is not None:
            self.business_name = business_name
        if customer_number is not None:
            self.customer_number = customer_number
        if reference_note is not None:
            self.reference_note = reference_note
        if sales_tax is not None:
            self.sales_tax = sales_tax
        if tax_exempt is not None:
            self.tax_exempt = tax_exempt
        if card_tracks is not None:
            self.card_tracks = card_tracks
        if card_zip is not None:
            self.card_zip = card_zip
        if card_address is not None:
            self.card_address = card_address
        if item_description is not None:
            self.item_description = item_description
        if default_transaction_type is not None:
            self.default_transaction_type = default_transaction_type

    @property
    def cvv(self):
        return self._cvv

    @cvv.setter
    def cvv(self, cvv):
        self._cvv = cvv

    @property
    def contact_name(self):
        return self._contact_name

    @contact_name.setter
    def contact_name(self, contact_name):
        self._contact_name = contact_name

    @property
    def contact_title(self):
        return self._contact_title

    @contact_title.setter
    def contact_title(self, contact_title):
        self._contact_title = contact_title

    @property
    def email_address(self):
        return self._email_address

    @email_address.setter
    def email_address(self, email_address):
        self._email_address = email_address

    @property
    def phone(self):
        return self._phone

    @phone.setter
    def phone(self, phone):
        self._phone = phone

    @property
    def fax(self):
        return self._fax

    @fax.setter
    def fax(self, fax):
        self._fax = fax

    @property
    def billing_address(self):
        return self._billing_address

    @billing_address.setter
    def billing_address(self, billing_address):
        self._billing_address = billing_address

    @property
    def billing_address2(self):
        return self._billing_address2

    @billing_address2.setter
    def billing_address2(self, billing_address2):
        self._billing_address2 = billing_address2

    @property
    def billing_city(self):
        return self._billing_city

    @billing_city.setter
    def billing_city(self, billing_city):
        self._billing_city = billing_city

    @property
    def billing_state(self):
        return self._billing_state

    @billing_state.setter
    def billing_state(self, billing_state):
        self._billing_state = billing_state

    @property
    def billing_zip(self):
        return self._billing_zip

    @billing_zip.setter
    def billing_zip(self, billing_zip):
        self._billing_zip = billing_zip

    @property
    def po_number(self):
        return self._po_number

    @po_number.setter
    def po_number(self, po_number):
        self._po_number = po_number

    @property
    def invoice_number(self):
        return self._invoice_number

    @invoice_number.setter
    def invoice_number(self, invoice_number):
        self._invoice_number = invoice_number

    @property
    def business_name(self):
        return self._business_name

    @business_name.setter
    def business_name(self, business_name):
        self._business_name = business_name

    @property
    def customer_number(self):
        return self._customer_number

    @customer_number.setter
    def customer_number(self, customer_number):
        self._customer_number = customer_number

    @property
    def reference_note(self):
        return self._reference_note

    @reference_note.setter
    def reference_note(self, reference_note):
        self._reference_note = reference_note

    @property
    def sales_tax(self):
        return self._sales_tax

    @sales_tax.setter
    def sales_tax(self, sales_tax):
        self._sales_tax = sales_tax

    @property
    def tax_exempt(self):
        return self._tax_exempt

    @tax_exempt.setter
    def tax_exempt(self, tax_exempt):
        self._tax_exempt = tax_exempt

    @property
    def card_tracks(self):
        return self._card_tracks

    @card_tracks.setter
    def card_tracks(self, card_tracks):
        self._card_tracks = card_tracks

    @property
    def card_zip(self):
        return self._card_zip

    @card_zip.setter
    def card_zip(self, card_zip):
        self._card_zip = card_zip

    @property
    def card_address(self):
        return self._card_address

    @card_address.setter
    def card_address(self, card_address):
        self._card_address = card_address

    @property
    def item_description(self):
        return self._item_description

    @item_description.setter
    def item_description(self, item_description):
        self._item_description = item_description

    @property
    def default_transaction_type(self):
        return self._default_transaction_type

    @default_transaction_type.setter
    def default_transaction_type(self, default_transaction_type):
        self._default_transaction_type = default_transaction_type
